static/app/views/dashboardsV2/detail.tsx

```diff
@@ -327,6 +332,11 @@ class DashboardDetail extends Component<Props, State> {
         break;
       }
       case DashboardState.EDIT: {
+        // TODO(nar): This should only fire when there are changes to the layout
+        if (organization.features.includes('dashboard-grid-layout')) {
+          saveDashboardLayout(organization.id, dashboard.id, this.state.layout);
+        }
```

Comment:
Also you probably already have this planned but I think we'll need to add layout saving to `DashboardState.CREATE` commits as well. The challenge here might be getting the widget ids for the layout array (dashboard id should be easily accessible through `newDashboard.id`)

Reply:
To summarize here, I have a ticket to track the work for create and new widgets and that's going to be my next set of work. I'm also going to punt on the issue to save the layout when validation passes until later. It will need to be addressed when converting to an API-source for sure.